x-pack/platform/plugins/shared/fleet/server/services/agents/migrate_action_runner.ts

```diff
@@ -82,8 +82,8 @@ export async function bulkMigrateAgentsBatch(
     }
   });
   const actionId = options.actionId ?? uuidv4();
-  const total = options.total ?? agents.length;
   const agentIds = agentsToAction.map((agent) => agent.id);
+  const total = agentIds.length; // Always use the actual agents being processed in this batch so that progress tracking is accurate
```

Comment:
I think we want to store the total agent count across all batches, like we do in all other actions.